packages/core/src/session_pool/session_pool.ts

```diff
@@ -367,7 +367,14 @@ export class SessionPool extends EventEmitter {
             persistStateKeyValueStoreId: this.persistStateKeyValueStoreId,
             persistStateKey: this.persistStateKey,
         });
-        await this.keyValueStore.setValue(this.persistStateKey, this.getState());
+
+        // use half the interval of `persistState` to avoid race conditions
+        const persistStateIntervalMillis = this.config.get('persistStateIntervalMillis')!;
+        const timeoutSecs = persistStateIntervalMillis / 2_000;
+        await this.keyValueStore.setValue(this.persistStateKey, this.getState(), {
+            timeoutSecs,
+            doNotRetryTimeouts: true,
```

Comment:
nit: how about `retryOnTimeouts:` or `retryTimeouts: false`?

Reply:
this is an existing option in the client, no plans to introduce a new name for it here, IMO that would make things only more confusing

(i don't like its name too, but still, lets stay consistent)